apis/github

```diff
@@ -1,19 +1,19 @@
 local JSON = dofile("apis/dkjson")
 
--- Build a github API url, trying to authenticate (not tested)
+local function getAuthHeaders(auth)
```

Comment:
I don't think this concern is separable, because if we implement `basic` authentication again, this function wouldn't be useful, and we'd still need to specialize getAPI. So I think this was better inside getAPI.